src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -753,7 +753,7 @@ private static ImmutableArray<ITestInfo> CreateTestCases(IMethodSymbol method, L
                             // Emit diagnostic
                             continue;
                         }
-                        var argsAsCode = ImmutableArray.CreateRange(args.Select(a => a.ToCSharpString()));
+                        var argsAsCode = ImmutableArray.CreateRange(args.Select(a => a.ToCSharpString() + (a.Type!.SpecialType == SpecialType.System_Single ? "F" : "")));
```

Comment:
Without this change, a float constant in [InlineData] leads to a build failure when trying to pass a double constant for a float parameter.  This isn't a great or complete solution, but I couldn't find a good public entry point in Roslyn to do this correctly, and float cases do not work at all currently.